tests/test_assets.py

```diff
@@ -966,6 +970,198 @@ def test_sids(self):
+                    'exchange': 'TEST',
+                },
+                {
+                    'sid': 2,
+                    'symbol': 'C',
+                    'start_date': pd.Timestamp('2013-7-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                    'exchange': 'TEST',
+                },
+            ]
+        )
+
+        supplementary_mappings = pd.DataFrame.from_records(
+            [
+                {
+                    'sid': 0,
+                    'field': 'ALT_ID',
+                    'value': '100000000',
+                    'start_date': pd.Timestamp('2013-1-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2013-6-28', tz='UTC'),
+                },
+                {
+                    'sid': 1,
+                    'field': 'ALT_ID',
+                    'value': '100000001',
+                    'start_date': pd.Timestamp('2013-1-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                },
+                {
+                    'sid': 0,
+                    'field': 'ALT_ID',
+                    'value': '100000002',
+                    'start_date': pd.Timestamp('2013-7-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                },
+                {
+                    'sid': 2,
+                    'field': 'ALT_ID',
+                    'value': '100000000',
+                    'start_date': pd.Timestamp('2013-7-1', tz='UTC'),
+                    'end_date': pd.Timestamp('2014-1-1', tz='UTC'),
+                },
+            ]
+        )
+
+        self.write_assets(
+            equities=equities,
+            supplementary_mappings=supplementary_mappings,
+        )
+
+        af = self.asset_finder
+
+        # Before sid 0 has changed ALT_ID.
+        dt = pd.Timestamp('2013-6-28', tz='UTC')
+
+        asset_0 = af.lookup_by_supplementary_field('ALT_ID', '100000000', dt)
+        self.assertEqual(asset_0.sid, 0)
+
+        asset_1 = af.lookup_by_supplementary_field('ALT_ID', '100000001', dt)
+        self.assertEqual(asset_1.sid, 1)
+
+        # We don't know about this ALT_ID yet.
+        with self.assertRaises(ValueNotFoundForField):
+            af.lookup_by_supplementary_field('ALT_ID', '100000002', dt)
+
+        # After all assets have ended.
+        dt = pd.Timestamp('2014-01-02', tz='UTC')
+
+        asset_2 = af.lookup_by_supplementary_field('ALT_ID', '100000000', dt)
+        self.assertEqual(asset_2.sid, 2)
+
+        asset_1 = af.lookup_by_supplementary_field('ALT_ID', '100000001', dt)
+        self.assertEqual(asset_1.sid, 1)
+
+        asset_0 = af.lookup_by_supplementary_field('ALT_ID', '100000002', dt)
+        self.assertEqual(asset_0.sid, 0)
+
+        # At this point both sids 0 and 2 have held this value, so an
+        # as_of_date is required.
+        with self.assertRaises(MultipleValuesFoundForField):
```

Comment:
can we add a test for the repr of the error message to make sure we are forwarding the proper info to the message?